src/coreclr/jit/redundantbranchopts.cpp

```diff
@@ -691,6 +692,307 @@ bool Compiler::optJumpThread(BasicBlock* const block, BasicBlock* const domBlock
     return true;
 }
 
+//------------------------------------------------------------------------
+// optRedundantRelop: see if the value of tree is redundant given earlier
+//   relops in this block.
```

Comment:
It might be useful if you give an example here of the type of transformations done

Reply:
Added one -- was that what you had in mind?

Reply:
That's great! Much more detailed even than what I imagined :-)